Update docu contribution guidelines (#831)
kyma-project/community: docs/guidelines/content-guidelines/05-tabs.md

---
title: Tabs & toggles
---

Tabs and toggles allow you to render several versions of a given section in one document or have several versions of one document on `http://kyma-project.io`.

> You can use tabs and/or documentation toggle to include instructions for different operating systems (OS) in one guide or to describe installation guides that differ only in a few steps.

Reviewer: ```suggestion
For example, you can use tabs and/or documentation toggle to include instructions for different operating systems (OS) in one guide, or to describe installation guides that differ only in a few steps.
```

> 

Reviewer: Can we add a sentence that explains what's the main difference between tabs and toggles (iirc, tabs are docsify-specific markdown, whereas toggles use html within markdown?), and maybe a screenshot how they look rendered? Just as a first pointer to support the decision what to use in a specific case.

## Docsify tabs

To display tabbed content from Markdown, kyma-project.io uses [docsify-tabs plugin](https://jhildenbiddle.github.io/docsify-tabs/#/) from [docsify.js](https://docsify.js.org/#/).

> You define a tab set using `tabs:start` and `tabs:end` HTML comments. For detailed usage guidelines see the [Usage section in the plugin official documentation](https://jhildenbiddle.github.io/docsify-tabs/#/?id=usage).

Reviewer: ```suggestion
You define a tab set using `tabs:start` and `tabs:end` HTML comments. For detailed guidelines, see the [Usage section in the plugin official documentation](https://jhildenbiddle.github.io/docsify-tabs/#/?id=usage).
```

> > **CAUTION:** The use of HTML comments prevents tab-related markup from being displayed when Markdown is rendered as HTML outside of `http://kyma-project.io`, for example, in GitHub. As a result in GitHub the content displays in sections under headers defined in the tab set.

Reviewer: ```suggestion
> **CAUTION:** The use of HTML comments prevents tab-related markup from being displayed when Markdown is rendered as HTML outside of `http://kyma-project.io`, for example, in GitHub. As a result, in GitHub, the content displays in sections under headers defined in the tab set.
```

## Documentation toggle

> 

Reviewer: This section describes that they look different on GitHub and on the website, right? Then I'd restructure the sentences to make this fact more prominent.

> The toggle renders the text in Github documents in the form of expandable sections and shows it as tabs on the `http://kyma-project.io` website.

Reviewer: ```suggestion
In Github documents, the toggle renders the text in the form of expandable sections.
On the `http://kyma-project.io` website, it's displayed as tabs; using a dedicated script.
```

> Content from documentation toggles is displayed as tabs on `http://kyma-project.io` using a dedicated script.

Reviewer: ```suggestion
```
redundant

### Writing rules

Follow these rules when inserting toggles in documents:

* Toggles belonging to one set must start with the `<div tabs name="{toggle-name}">` tag and end with the `</div>` tag, where **name** is a distinctive ID of a specific toggle used for linking. It can be any name describing the content of the documentation toggle.

> See the example:

Reviewer: as all of this is part of a bullet list, the examples should be indented with (probably) two blank spaces so that they're rendered below the respective list item.

> ```markdown

Reviewer: Adding blank spaces
Also, isn't that html rather than markdown?

```suggestion
  ```html
```

> <div tabs name="installation">

Reviewer: ```suggestion
  <div tabs name="installation">
```

> {toggle content}

Reviewer: ```suggestion
  {toggle content}
```

> </div>

Reviewer: ```suggestion
  </div>
```

> ```

Reviewer: ```suggestion
  ```
```

* A single toggle must start with the `<details>` tag and end with the `</details>` tag.

> * Insert the title of the toggle between `<summary>` and `</summary>` tags. Every part of the title must start from a new line.

Reviewer: ```suggestion
* Insert the title of the toggle between the `<summary>` and `</summary>` tags. Every part of the title must start from a new line.
```

> See the example:

Reviewer: Add 2 blank spaces indentation in the following lines

``` markdown
<summary>
From release
</summary>
```

> * Leave a blank line after the `</summary>` tag. Only this way the content can render.

Reviewer: ```suggestion
* Leave a blank line after the `</summary>` tag. Only this way, the content can render.
```

* Add content using Markdown syntax.

> * Use this indentation when you add the toggle as a regular document section:

Reviewer: ```suggestion
* Use the following indentation when you add the toggle as a regular document section:
```

  - No spaces for the div element
  - Two spaces for other elements

> * Use this indentation when you add a toggle under a list:

Reviewer: ```suggestion
* Use the following indentation when you add a toggle under a list:
```

  - Four spaces for the div element
  - Six spaces for other elements

> * When you add two or more toggle sets that belong to the same group, you must classify them properly. This way each time you switch to one of the toggles in the set, it opens automatically in all other sets from the same group within one document and across all documents that use the group. To create a toggle group:

Reviewer: ```suggestion
* When you add two or more toggle sets that belong to the same group, you must classify them properly. This way each, time you switch to one of the toggles in the set, it opens automatically in all other sets from the same group within one document and across all documents that use the group. To create a toggle group:
```

Reviewer: Btw this is a super cool feature, I'm impressed!

  - All sets of toggles from the given group must contain the `group` attribute with a distinctive ID of this group. For example, add the `cluster-installation` group to categorize all toggle sets that refer to Kyma installation on a cluster.
  - Every `<summary>` tag in the set must also contain the `label` attribute. For example, add the `GKE` label to all toggles that describe steps for Kyma installation on a GKE cluster in the `cluster-installation` group.

    > **NOTE:** The label does not have to be the same as the summary. If the label is a combination of two or more words, separate them with the dash (`-`). For example, write `<summary label="install-on-gke">`.

>   See the example:

Reviewer: Does this example refer to the bullet point above; or to the entire toggle section?
If it belongs to the specific bullet point, we need more blanks so it's indented below the bullet point.
If it's an example for a toggle section, we could state that explicitly.

Author: It goes to the "* When you add two or more toggle sets that belong to the same group, you must classify them properly. This way each, time you switch to one of the toggles in the set, it opens automatically in all other sets from the same group within one document and across all documents that use the group. To create a toggle group:" bullet point. It seems to render ok for me but I may be missing something.

  ``` markdown
  <div tabs name="prerequisites" group="cluster-installation">
    <details>
    <summary label="gke">
    GKE
    </summary>
    ...
    </details>
    <details>
    <summary label="aks">
    AKS
    </summary>
    ...
    </details>
    <details>
    <summary label="gardener">
    Gardener
    </summary>
    ...
    </details>
  </div>
  ```

> ### Examples

Reviewer: Oh, now I realize the example above probably only belongs to the one bullet point :D

This section contains exemplary uses of the toggle in Markdown documents.

* See the example of the toggle added as a regular document section:

> ```markdown

Reviewer: Isn't that html rather than markdown?

```suggestion
```html
```

   - **Subscription name**: `lastorder-sub`
   - **Config**: `maxInFlightMessages: 5`
   - **Types**: `order.received.v1`
   - **Service**: `lastorder` (The sink field will be populated automatically.)
   - **Type matching:**: `standard`
   - **Source**: `myapp`
     
5. Click **Create**.
6. Wait a few seconds for the Subscription to have status `READY`.

  </details>
  <details>
  <summary label="kubectl">
  kubectl
  </summary>

Run:
```bash
cat <<EOF | kubectl apply -f -
   apiVersion: eventing.kyma-project.io/v1alpha2
   kind: Subscription
   metadata:
     name: lastorder-sub
     namespace: default
   spec:
     config:
       maxInFlightMessages: "5"
     sink: 'http://lastorder.default.svc.cluster.local'
     source: myapp
     types:
       - order.received.v1
EOF
```

To check that the Subscription was created and is ready, run:
```bash
kubectl get subscriptions lastorder-sub -o=jsonpath="{.status.ready}"
```

The operation was successful if the returned status says `true`.
  </details>
</div>
```

The code renders on `kyma-project.io` as follows:

![](./assets/toggle-flat-structure.png)

* See the example of the toggle added under a list:

> ```markdown

Reviewer: Isn't that html rather than markdown?

```suggestion
```html
```
